src/components/dropdown.component.js

```diff
@@ -4,15 +4,51 @@ export default class DropdownComponent {
     constructor(query) {
         this.query = query;
 
+        this.dropdownContainer = $(`${query} .dropdown-container`);
         this.btn = $(`${query} .dropdown-btn`);
-        this.items = $(`${query} .dropdown-item`);
         this.list = $(`${query} .dropdown-list`);
+        this.items = $(`${query} .dropdown-item`);
         this.selectedItem = $(`${query} .selected`);
 
+        this.enabled = true;
+
         this.addDropdownClickEventListner();
         this.addItemClickEventListner();
     }
 
+    getSelectedItem() {
+        return this.selectedItem.text();
+    }
+
+    checkQuery() {
+        if (
+            this.query.length === 0 ||
+            this.dropdownContainer.length === 0 ||
+            this.btn.length === 0 ||
+            this.list.length === 0 ||
+            this.items.length === 0
+        ) {
+            return false;
+        }
+        return true;
+    }
+
+    setEnabled() {
+        this.enabled = this.dropdownContainer.prop('disabled', false);
```

Comment:
The prop method does not return a boolean value:
![image](https://user-images.githubusercontent.com/18483175/161963422-6fb1869e-b771-44bd-9493-5d1ce22151dd.png)

You are assing a no boolean value to a boolean property. Even JS is a dynamic language this is not a good approach, because your variable does not correspond to his value.

Reply:
Used t"his.disabled = false" now, to change the value, disable method do "this.disabled = this.container.is(':disabled');" and enable do the same.